syntax = "proto3";

package v1;

import "google/protobuf/wrappers.proto";
import "v1/service.proto";
import "v1/routing.proto";
import "v1/client.proto";
import "v1/ratelimit.proto";
import "v1/circuitbreaker.proto";

message Response {
	google.protobuf.UInt32Value code = 1;
	google.protobuf.StringValue info = 2;
	Client client = 3;
	//Namespace namespace = 4;
	Service service = 5;
	Instance instance = 6;
	Routing routing = 7;
	//ServiceAlias alias = 8;
	Rule rateLimit = 9;
	CircuitBreaker circuitBreaker = 10;
	ConfigRelease configRelease = 11;
}

message ConfigRelease {
  Service service = 1;
  google.protobuf.StringValue ctime = 2;
  google.protobuf.StringValue mtime = 3;
  
  CircuitBreaker circuitBreaker = 4;
}

message BatchWriteResponse {
	google.protobuf.UInt32Value code = 1;
	google.protobuf.StringValue info = 2;
	google.protobuf.UInt32Value size = 3;
	repeated Response responses = 4;
}

message DiscoverResponse {
	google.protobuf.UInt32Value code = 1;
	google.protobuf.StringValue info = 2;

	enum DiscoverResponseType {
		UNKNOWN = 0;
		INSTANCE = 1;
		CLUSTER = 2;
		ROUTING = 3;
		RATE_LIMIT = 4;
		CIRCUIT_BREAKER = 5;
	}

	DiscoverResponseType type = 3;
	Service service = 4;
	repeated Instance instances = 5;
	Routing routing = 6;
	RateLimit rateLimit = 7;
	CircuitBreaker circuitBreaker = 8;
}

message StatResponse {
	google.protobuf.StringValue id = 1; // 唯一标识
	google.protobuf.UInt32Value code = 2; // 返回码
	google.protobuf.StringValue info = 3; // 返回信息
}